### ISSUE TITLE
Is map_struct_to_json thread safe?

### ISSUE THREAD

**Issue by byrnexu** [bk192077/struct_mapping#11]
The following code sometimes gets the first half of json, I suspect it is because of the result of calling in a multi-threaded environment, the following fieldinfo is a local variable

   struct_mapping::map_struct_to_json(fieldInfo, oss, "  ");

**Comment by bk192077** [comment]
No, struct_mapping is not thread safe. But perhaps that's not the point. It would be nice to see a more detailed code example.
